src/pieces/pawn.cpp

```diff
@@ -0,0 +1,236 @@
+#include "pieces/Pawn.h"
+#include "Board.h"
+
+using Chess::PawnPiece;
+using Chess::Movement;
+using Chess::utilities::Color;
+using Chess::utilities::PieceType;
+using Chess::utilities::Direction;
+using Chess::utilities::DirectionOffset;
+
+PawnPiece::PawnPiece(Coordinate coordinate, Color color, PieceType type) : Piece{coordinate,color,type}
+{
+    symbol = (color == Color::black) ? 'P': 'p';
+}
+std::list<Movement> PawnPiece::get_pseudo_valid_movements(Board& board)
+{
+    std::list<Movement> pseudo_moviments;
+    Coordinate test_coordinate;
+    std::pair<int,int> offset; 
+    std::shared_ptr<Piece> test_piece;
+    //if it's the pawn's first move, it can move two squares straight forward
```

Comment:
forse ti conviene raggruppare tutto il due macrocategorie: bianco, con le tre posizioni davanti + eventuale doppio, lo stesso con il nero. In questo modo credo che il codice sarà più leggibile, puoi anche creare una funzione separata, dato che bianco e nero sono simmetrici